src/ejabberd_c2s.erl

```diff
@@ -1759,6 +1768,11 @@ handle_info({broadcast, Type, From, Packet}, StateName, StateData) ->
     fsm_next_state(StateName, StateData);
 handle_info(dont_ask_offline, StateName, StateData) ->
     fsm_next_state(StateName, StateData#state{ask_offline = false});
+handle_info(close, StateName, StateData) ->
+    ?DEBUG("Timeout waiting for stream management acknowledgement of ~s",
+	   [jid:to_string(StateData#state.jid)]),
+    close(self()),
+    fsm_next_state(StateName, StateData);
```

Comment:
> But we could of course set `mgmt_ack_timer = undefined` to clarify the code.

I was going to update the PR accordingly, but there's the minor downside that a new timer would be spawned if another stanza arrives before the process actually goes into the `wait_for_resume` state.  This is avoided by *not* setting `mgmt_ack_timer = undefined`.  We could set it to `expired` or something, though :-)